src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,154 @@
     public T get(int index) {
-        return null;
+        Node<T> current = findNodeByIndex(index);
+        return current.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = findNodeByIndex(index);
+        T oldValue = current.item;
+        if (index == 0) {
+            head.item = value;
+        } else {
+            current.item = value;
+        }
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = findNodeByIndex(index);
+        return unlike(current);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if (object == null) {
+                if (current.item == null) {
+                    unlike(current);
+                    return true;
+                }
+            } else {
+                if (object.equals(current.item)) {
+                    unlike(current);
+                    return true;
+                }
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private void checkIndexForAdd(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of bounds.");
+        }
+    }
+
+    private void checkIndex(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index out of bounds.");
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        checkIndex(index);
+        Node<T> node;
+        if (index < size / 2) {
+            node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+        } else {
+            node = tail;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+        }
+        return node;
+    }
+
+    private T unlike(Node<T> node) {
+        final T oldValue = node.item;
+        if (node == head) {
+            head = node.next;
+        } else {
+            node.prev.next = node.next;
+        }
+        if (node == tail) {
+            tail = node.prev;
+        } else {
+            node.next.prev = node.prev;
+        }
+        size--;
+        return oldValue;
```

Comment:
In the `unlink` method, you should check if the `next` and `prev` nodes are not null before unlinking the node. This will prevent `NullPointerException`.